src/NuGet.Services.EndToEnd/Support/TestSettings.cs

```diff
@@ -121,9 +121,17 @@ private static async Task<TestSettings> CreateInternalAsync()
 
         private static async Task<TestSettings> CreateInternalAsync(string configurationName)
         {
+            var configurationDirectoryName = Path.Combine(Environment.CurrentDirectory, "config");
+            var configurationFilename = configurationName + ".json";
+            var configurationFile = new FileInfo(Path.Combine(configurationDirectoryName, configurationFilename));
+            if (!configurationFile.Exists)
+            {
+                throw new InvalidOperationException($"Configuration file not found: {configurationFile.FullName}");
+            }
+
             var builder = new ConfigurationBuilder()
-             .SetBasePath(Path.Combine(Environment.CurrentDirectory, "config"))
-             .AddJsonFile(configurationName + ".json");
+                .SetBasePath(configurationDirectoryName)
+                .AddJsonFile(configurationFilename);
```

Comment:
this should have a "required" parameter that does the same thing, right?

Reply:
Yeah, it would be best if we didn't have to construct the path ourselves to verify if the file exists.

[There appears to be an `optional` parameter.](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.jsonconfigurationextensions.addjsonfile?view=aspnetcore-2.2#Microsoft_Extensions_Configuration_JsonConfigurationExtensions_AddJsonFile_Microsoft_Extensions_Configuration_IConfigurationBuilder_System_String_System_Boolean_) Have you tried that?